test/AccessLock.t.sol

```diff
@@ -172,7 +194,30 @@ contract AccessLockTest is Test, Deployers {
         assertEq(MockERC20(Currency.unwrap(currency1)).balanceOf(address(accessLockHook)), amount);
     }
 
-    function test_beforeModifyPosition_swap_succeedsWithAccessLock() public {
+    function test_beforeRemoveLiquidity_take_succeedsWithAccessLock() public {
+        // Add liquidity so there is something to take.
+        delta = modifyPositionRouter.modifyPosition(key, LIQ_PARAMS, ZERO_BYTES);
+        // Can't take more than the manager has.
+        uint128 takeAmount = uint128(key.currency1.balanceOf(address(manager)) - 1);
+
+        uint256 balanceOfBefore1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+        uint256 balanceOfBefore0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+
+        // Hook only takes currency 1 rn.
+        modifyPositionRouter.modifyPosition(
+            key, REMOVE_LIQ_PARAMS, abi.encode(takeAmount, AccessLockHook.LockAction.Take)
+        );
+        uint256 balanceOfAfter0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+        uint256 balanceOfAfter1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+
+        // The balance of our contract should be equal to our original balance because we added and removed our liquidity.
+        // TODO: Maybe due to a rounding issue, but balanceOfBefore0 == balanceOfAfter0 ?
+        assertTrue(balanceOfBefore0 + uint256(uint128(delta.amount0())) - balanceOfAfter0 <= 1);
```

Comment:
I'd write these tests to test equality and just let them fail for now until we figure out why its off by one

Reply:
Turns out this is a known issue see [here](https://github.com/Uniswap/v3-core/issues/570).